functions/java-jre.bash

```diff
@@ -211,10 +213,10 @@ java_zulu_fetch() {
   if [[ -z $downloadLink ]]; then echo "FAILED (download link)"; return 1; fi
 
   if ! mkdir -p "$jdkInstallLocation"; then echo "FAILED (create directory)"; return 1; fi
-  if ! rm -rf "${jdkInstallLocation:?}"/*; then echo "FAILED (clean directory)"; return 1; fi
-  if ! cond_redirect wget -nv -O "$jdkInstallLocation"/zulu.tar.gz "$downloadLink"; then echo "FAILED (download)"; rm -rf "${jdkInstallLocation:?}"/*; return 1; fi
-  if ! cond_redirect tar -xpzf "$jdkInstallLocation"/zulu.tar.gz -C "$jdkInstallLocation"; then echo "FAILED (extract)"; rm -rf "${jdkInstallLocation:?}"/*; return 1; fi
-  if cond_redirect rm -rf "$jdkInstallLocation"/zulu.tar.gz; then echo "OK"; else echo "FAILED (cleanup)"; return 1; fi
+  if ! cond_redirect wget -nv -O "$temp" "$downloadLink"; then echo "FAILED (download)"; rm -f "$temp"; return 1; fi
+  if ! cond_redirect tar -xpzf "$temp" -C "$jdkInstallLocation"; then echo "FAILED (extract)"; rm -rf "$(find "$jdkInstallLocation" -maxdepth 1 -type d -printf '%T@\t%p\n' | sort | tail -n 1 | sed 's/[0-9]*\.[0-9]*\t//')"; rm -f "$temp"; return 1; fi
+  if ! rm -rf "$(find "$jdkInstallLocation" -maxdepth 1 -type d -printf '%T@\t%p\n' | sort -r | tail -n 1 | sed 's/[0-9]*\.[0-9]*\t//')"; then echo "FAILED (clean directory)"; return 1; fi
```

Comment:
Err I think I'm not quite getting what this does. You untar first then remove from that dir you just unpacked to ?

Reply:
No, I'm packing it then using the find command to delete everything except what I just extracted.